connection.go

```diff
@@ -0,0 +1,596 @@
+	}
+
+	c.BrokerConnection = nc
+	c.JSContext = js
+
+	natsConnectionID, err := c.generateNatsConnectionID()
+	if err != nil {
+		return fmt.Errorf("superstream: error connecting to superstream: %v", err)
+	}
+	c.NatsConnectionID = natsConnectionID
+
+	return nil
+}
+
+func (c *Client) RegisterClient() error {
+	registerReq := RegisterReq{
+		NatsConnectionID: c.NatsConnectionID,
+		Language:         "go",
+		Version:          sdkVersion,
+		LearningFactor:   c.LearningFactor,
+		Config:           c.Config,
+	}
+
+	registerReqBytes, err := json.Marshal(registerReq)
+	if err != nil {
+		return fmt.Errorf("superstream: error registering client: %v", err)
+	}
+
+	resp, err := c.BrokerConnection.Request(clientRegisterSubject, registerReqBytes, 30*time.Second)
+	if err != nil {
+		return fmt.Errorf("superstream: error registering client: %v", err)
+	}
+
+	var registerResp RegisterResp
+	err = json.Unmarshal(resp.Data, &registerResp)
+	if err != nil {
+		return fmt.Errorf("superstream: error registering client: %v", err)
+	}
+
+	c.ClientID = registerResp.ClientID
+	c.AccountName = registerResp.AccountName
+	c.LearningFactor = registerResp.LearningFactor
+	c.LearningFactorCounter = 0
+	c.LearningRequestSent = false
+	c.GetSchemaRequestSent = false
+	c.ConsumerProtoDescMap = make(map[string]protoreflect.MessageDescriptor)
+	c.IsConsumer = false
+	c.IsProducer = false
+	c.Counters = ClientCounters{
+		TotalBytesBeforeProduce:           0,
+		TotalBytesAfterProduce:            0,
+		TotalBytesBeforeConsume:           0,
+		TotalBytesAfterConsume:            0,
+		TotalMessagesSuccessfullyProduce:  0,
+		TotalMessagesSuccessfullyConsumed: 0,
+		TotalMessagesFailedProduce:        0,
+		TotalMessagesFailedConsume:        0,
+	}
+
+	return nil
+}
+
+func (c *Client) SubscribeUpdates() error {
+	cus := ClientUpdateSub{
+		ClientID:   c.ClientID,
+		UpdateCahn: make(chan Update),
+	}
+
+	go cus.UpdatesHandler()
+
+	var err error
+	cus.Subscription, err = c.BrokerConnection.Subscribe(fmt.Sprintf(superstreamClientUpdatesSubject, c.ClientID), cus.SubscriptionHandler())
+	if err != nil {
+		return fmt.Errorf("superstream: error connecting to superstream %v", err)
+	}
+
+	return nil
+}
+
+func (c *ClientUpdateSub) UpdatesHandler() {
+	for {
+		msg := <-c.UpdateCahn
+		switch msg.Type {
+		case "LearnedSchema":
+			var schemaUpdateReq SchemaUpdateReq
+			err := json.Unmarshal(msg.Payload, &schemaUpdateReq)
+			if err != nil {
+				handleError(fmt.Sprintf("[sdk: go][version: %v]UpdatesHandler at json.Unmarshal: %v", sdkVersion, err.Error()))
+			}
+			desc := compileMsgDescriptor(schemaUpdateReq.Desc, schemaUpdateReq.MasterMsgName, schemaUpdateReq.FileName)
+			if desc != nil {
+				ClientConnection.ProducerProtoDesc = desc
+				ClientConnection.ProducerSchemaID = schemaUpdateReq.SchemaID
+			} else {
+				handleError(fmt.Sprintf("[sdk: go][version: %v]UpdatesHandler: error compiling schema ", sdkVersion))
```

Comment:
the sdk version can be set in the function handleError itself you don't need to send it everytime you are calling to this function